### DISCUSSION TITLE
Struggles with wiggle keying and removing it from some bones

### DISCUSSION THREAD

**Discussion by not-that-nda** [shteeve3d/blender-wiggle-2#29] [General]
First of all, thanks a lot for this excellent addon, I love it and use it daily on animations for games.

I have some issues when I accidentally set the wiggle properties for some bones (for some reason, I added it to the IK leg poles), and then it is impossible to make it go away. The custom properties are set by the API, and I don't know how to clean them up.

Another issue I face is when there are some of the values that are keyed for a bone, it is very hard to make them stay as I wanted them. Maybe implement some kind of script to clear all the keys for the current action on all selected bones, or clear the wiggle entirely from the selected bones so I can get rid once and for all of the wiggle from those knee pole targets.

A much-needed feature is, in my eyes, the addition of presets that you can set and save so you can easily assign properties to bones again from a configuration you saved without having to change each of the values.

Thanks a lot for all the work on this, it is awesome!

Cheers,
Nacho

**Comment by shteeve3d** [comment]
hey thanks for the feedback! i'm curious if you can provide some simple example file of the api set properties getting stuck. i know this was a problem on the first version of the wiggle addon that i ran into myself, but i tried to set things up in a way that should avoid it in the wiggle 2 version!

it might be related to your previous point, but you aren't able to clear the wiggle from a knee pole target just by disabling the wiggle on that bone?

lastly, agreed on some sort of preset system. its a blender api feature i have yet to learn but should!

**Comment by not-that-nda** [comment]
Apologies I have not gotten around to doing this yet. I have it in my todo list but I have not been animating bone wiggles much in the last two months so I did not have the chance to do it. However I can tell you that there are a bunch of custom properties laying auround in a bunch ob bones in my armature and having a way to visualize them or delete, or reset, or something, could be a valuable improvement so I can remove them once and for all from the rig parts that dont need them.

I will let you know as soon as I encounter the issue again and provide some examples.
